core/templates/dev/head/pages/exploration-player-page/services/learner-answer-info.service.ts

```diff
@@ -0,0 +1,127 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for learner answer info.
+ */
+
+require('domain/statistics/statistics-domain.constants.ts');
+require(
+  'pages/exploration-player-page/services/answer-classification.service.ts');
+require('pages/exploration-player-page/services/exploration-engine.service.ts');
+require('pages/exploration-player-page/services/player-transcript.service.ts');
+
+
+var oppia = require('AppInit.ts').module;
+
+oppia.factory('LearnerAnswerInfoService', [
+  'AnswerClassificationService', 'ExplorationEngineService',
+  'PlayerTranscriptService', 'INTERACTION_IDS_WITHOUT_ANSWER_DETAILS',
+  'PROBABILITY_INDEXES',
+  function(
+      AnswerClassificationService, ExplorationEngineService,
+      PlayerTranscriptService, INTERACTION_IDS_WITHOUT_ANSWER_DETAILS,
+      PROBABILITY_INDEXES) {
+    var submittedAnswerInfoCount = 0;
+    var actualProbabilityIndex = null;
+    var randomProbabilityIndex = null;
+    var expId = null;
+    var exploration = null;
+    var state = null;
+    var stateName = null;
+    var interactionId = null;
+    var defaultOutcome = null;
+    var currentAnswer = null;
+    var currentInteractionRulesService = null;
+    var canAskLearnerForAnswerInfo = false;
+    var visitedStates = [];
+
+
+    var getRandomProbabilityIndex = function() {
+      var min = 0;
+      var max = 100;
+      return (Math.floor(Math.random() * (max - min + 1) ) + min) / 100;
+    };
+
+    return {
+      askLearnerForAnswerInfo: function(
+          answer, interactionRulesService) {
+        currentAnswer = answer;
+        currentInteractionRulesService = interactionRulesService;
+        expId = ExplorationEngineService.getExplorationId();
+        exploration = ExplorationEngineService.getExploration();
+        stateName = PlayerTranscriptService.getLastStateName();
+        state = exploration.getState(stateName);
+        interactionId = state.interaction.id;
+        defaultOutcome = state.interaction.defaultOutcome;
+
+        if (submittedAnswerInfoCount === 2) {
+          return;
+        }
+
+        if (!(state.solicitAnswerDetails)) {
+          return;
+        }
+
+        if (INTERACTION_IDS_WITHOUT_ANSWER_DETAILS.indexOf(
+          interactionId) !== -1) {
+          return;
+        }
+
+        if (visitedStates.indexOf(stateName) !== -1) {
+          return;
+        }
+
+        visitedStates.push(stateName);
+
+        var classificationResult = (
+          AnswerClassificationService.getMatchingClassificationResult(
+            stateName, state.interaction, answer,
+            interactionRulesService));
+        var outcome = classificationResult.outcome;
+
+        randomProbabilityIndex = getRandomProbabilityIndex();
+        /* eslint-disable dot-notation */
+        if (outcome !== defaultOutcome) {
```

Comment:
Ditto here. Isn't the highest probability to ask for learner answer info when the outcome is the default outcome?

Reply:
Done